rust/src/delta_arrow.rs

```diff
@@ -205,12 +200,10 @@ impl TryFrom<&ArrowField> for schema::SchemaField {
             arrow_field.is_nullable(),
             arrow_field
                 .metadata()
-                .as_ref()
-                .map_or_else(HashMap::new, |m| {
-                    m.iter()
-                        .map(|(k, v)| (k.clone(), serde_json::Value::String(v.clone())))
-                        .collect()
-                }),
+                .to_owned()
```

Comment:
i think we can avoid the `to_owned` call here since we are cloning the elements already inside the map.